### PR TITLE
[CSDK-280] - App logging out automatically when there is no network

### PR DESCRIPTION
**Pull request by chandrasekhar-plivo**

### CODE REVIEW

File: SwiftVoiceCallingApp/SwiftVoiceCallingApp/Controller/PlivoCallController.swift [hunk 1]
```diff
@@ -162,15 +162,15 @@ class PlivoCallController: UIViewController, CXProviderDelegate, CXCallObserverD
...
-            UtilClass.setUserAuthenticationStatus(false)
-            UserDefaults.standard.removeObject(forKey: kUSERNAME)
-            UserDefaults.standard.removeObject(forKey: kPASSWORD)
-            UserDefaults.standard.synchronize()
-            let _mainStoryboard = UIStoryboard(name: "Main", bundle: nil)
-            let loginVC: LoginViewController? = _mainStoryboard.instantiateViewController(withIdentifier: "LoginViewController") as? LoginViewController
-            Phone.sharedInstance.setDelegate(loginVC!)
-            let _appDelegate: AppDelegate? = (UIApplication.shared.delegate as? AppDelegate)
-            _appDelegate?.window?.rootViewController = loginVC
+//            UtilClass.setUserAuthenticationStatus(false)
```
Comment (anand-plivo) [inline SwiftVoiceCallingApp/SwiftVoiceCallingApp/Controller/PlivoCallController.swift]: it should not logout only when registration fails without internet connection, for other error cases we should logout

**Review by anand-plivo — CHANGES_REQUESTED** [review]

File: SwiftVoiceCallingApp/SwiftVoiceCallingApp/Controller/PlivoCallController.swift (hunk 1 above)
Comment (chandrasekhar-plivo) [reply SwiftVoiceCallingApp/SwiftVoiceCallingApp/Controller/PlivoCallController.swift]: check now

File: SwiftVoiceCallingApp/SwiftVoiceCallingApp/Controller/PlivoCallController.swift [hunk 2]
```diff
@@ -162,15 +162,28 @@ class PlivoCallController: UIViewController, CXProviderDelegate, CXCallObserverD
...
-//            UserDefaults.standard.removeObject(forKey: kPASSWORD)
-//            UserDefaults.standard.synchronize()
-//            let _mainStoryboard = UIStoryboard(name: "Main", bundle: nil)
-//            let loginVC: LoginViewController? = _mainStoryboard.instantiateViewController(withIdentifier: "LoginViewController") as? LoginViewController
-//            Phone.sharedInstance.setDelegate(loginVC!)
-//            let _appDelegate: AppDelegate? = (UIApplication.shared.delegate as? AppDelegate)
-//            _appDelegate?.window?.rootViewController = loginVC
+            
+            switch (error! as NSError).code {
+            case 502:
```
Comment (anand-plivo) [inline SwiftVoiceCallingApp/SwiftVoiceCallingApp/Controller/PlivoCallController.swift:167]: If we are sending this from SDK why don't we just stop this in our SDK itself.

**Review by anand-plivo — CHANGES_REQUESTED** [review]

File: SwiftVoiceCallingApp/SwiftVoiceCallingApp/Controller/PlivoCallController.swift (hunk 2 above)
Comment (chandrasekhar-plivo) [reply SwiftVoiceCallingApp/SwiftVoiceCallingApp/Controller/PlivoCallController.swift:167]: Along with pjsip 2.8 this change will be added to sdk

Comment (anand-plivo) [reply SwiftVoiceCallingApp/SwiftVoiceCallingApp/Controller/PlivoCallController.swift:167]: If this is a simple change on our SDK we should do this first then upgrade to 2.8

**Review by anand-plivo — APPROVED** [review]

File: SwiftVoiceCallingApp/SwiftVoiceCallingApp/Controller/PlivoCallController.swift (hunk 2 above)
Comment (chandrasekhar-plivo) [reply SwiftVoiceCallingApp/SwiftVoiceCallingApp/Controller/PlivoCallController.swift:167]: ok, noted

**Pull request merged**